Sources/Testing/Traits/Trait.swift

```diff
@@ -72,43 +192,14 @@ extension Trait {
   }
 }
 
+extension Trait where TestScopeProvider == Never {
```

Comment:
These extensions will show up (at least minimally) in DocC, so we might want to consider formally documenting them with full markup since they have some interesting behaviours.

(Not a blocker for the PR, just noting.)